### PR TITLE
Add some tests for gen1-3 abilities

### PR DESCRIPTION
**Pull request by SBird1337**
## Description
Adds tests for the following abilities:

 * Damp ✅
 * Intimidate ❌ [Bug: #2619 ]
 * Speed Boost ✅
 * Volt Absorb ✅
 * Insomnia ✅
 * Stench ✅ [Missing: Multi Hit interaction due to RNG]
 * Drizzle ✅
 * Limber ✅
 * Sand Veil ✅
 * Water Absorb ✅
 * Oblivous ✅
 * Cloud Nine ✅

Tests currently failing due to issues with the test system:

 * Compound Eyes: Fails due to a statistical anomaly when checking against OHKO moves.

## **Discord contact info**
Karathan#1337

### CODE REVIEW

**Review by mrgriffin — CHANGES_REQUESTED** [review]
This is awesome, thanks! :heart:

Some stylistic notes:
- I would try to rephrase the Intimidate tests to start with the word "Intimidate".
- `NONE_OF` should be more like `NONE_OF { x; }` rather than `NONE_OF(x);`.
- `PASSES_RANDOMLY` should occur before `GIVEN`. This one maybe we should add a check for, the reason is that `GIVEN` (and the other blocks) are not executed when determining the cost of a test for scheduling, so they will be erroneously counted as 1 battle.

But it's really good work :)

I'm curious about that Bulbapedia Speed Boost one. If I remember tomorrow maybe I can check what happens on Showdown.

**Comment by LOuroboros** [comment]
Dropping my tests for `ability_intimidate.c`, in case you want to add them:
```c
SINGLE_BATTLE_TEST("PlayerIntimidateOnSwitchOut")
{
    GIVEN
    {
        PLAYER(SPECIES_EKANS) { Ability(ABILITY_INTIMIDATE); }
        OPPONENT(SPECIES_MAGIKARP);
    }
    WHEN
    {
        TURN {}
    }
    SCENE
    {
        ABILITY_POPUP(player, ABILITY_INTIMIDATE);
        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
        MESSAGE("Ekans's Intimidate\ncuts Foe Magikarp's ATTACK!");
    }
}

SINGLE_BATTLE_TEST("EnemyIntimidateOnSwitchOut")
{
    GIVEN
    {
        PLAYER(SPECIES_MAGIKARP);
        OPPONENT(SPECIES_EKANS) { Ability(ABILITY_INTIMIDATE); }
    }
    WHEN
    {
        TURN {}
    }
    SCENE
    {
        ABILITY_POPUP(opponent, ABILITY_INTIMIDATE);
        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
        MESSAGE("Foe Ekans's Intimidate\ncuts Magikarp's ATTACK!");
    }
}
```

**Comment by SBird1337** [comment]
I'll have to think about / integrate the more verbose intimidate tests. For now I've addressed @mrgriffin's remarks. Please don't merge before squashing / ping me so I can squash it myself.

File: test/ability_damp.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_DAMP);
+        NONE_OF { HP_BAR(player); HP_BAR(opponent); }
+    }
+}
+
+// Bug: The ability popups don't show up.
+SINGLE_BATTLE_TEST("Damp prevents damage from aftermath")
+{
+    KNOWN_FAILING;
```
Comment (DizzyEggg) [inline test/ability_damp.c:43]: PASS #2645

File: test/ability_volt_absorb.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    } SCENE {
+        NONE_OF { ABILITY_POPUP(player, ABILITY_VOLT_ABSORB); HP_BAR(player); MESSAGE("Jolteon restored HP using its Volt Absorb!"); }
+    }
+}
+
+// This fails because the ability popup does not show
+// and Jolteon is not healed.
+SINGLE_BATTLE_TEST("Volt Absorb activates on status moves")
+{
+    KNOWN_FAILING;
```
Comment (DizzyEggg) [inline test/ability_volt_absorb.c:38]: PASS #2635

**Review by DizzyEggg — CHANGES_REQUESTED** [review]

**Comment by SBird1337** [comment]
I've revisited Speed Boost on Showdown which acted as Ground Truth for some of the more weird edge cases. The test would also fail there, so I removed it and call it weird phrasing from Bulbapedia.

I've rebased and squashed my commits, ready for another review :)

**Review by DizzyEggg — APPROVED** [review]

File: test/ability_cloud_nine.c [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_SANDSTORM].effect == EFFECT_SANDSTORM);
+        PLAYER(SPECIES_PSYDUCK) { Ability(ABILITY_CLOUD_NINE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SANDSTORM); }
+        TURN {}
+    } SCENE {
+        NONE_OF { HP_BAR(player); }
```
Comment (mrgriffin) [inline test/ability_cloud_nine.c:14]: You might like to check that the sandstorm move and/or end-of-turn animation doesn't play?

**Review by mrgriffin — APPROVED** [review]
Looking good, thanks! :heart:

File: test/ability_cloud_nine.c (hunk 3 above)
Comment (SBird1337) [reply test/ability_cloud_nine.c:14]: I think if I'm reading the ability correctly on bulbapedia the animations should still play (and they do) - the effects are just negated.
